R/base.R

```diff
@@ -65,7 +65,7 @@ with_no_new_device <- function(..., fun) {
   dev_after <- dev.cur()
   if (dev_before != dev_after) {
     dev.off(dev_after)
-    dev.set(dev_before)
+    if (dev_before > 1) dev.set(dev_before)
```

Comment:
`dev.set()` doesn't do what you'd expect it to when `dev_before` is the `null device`. Here's a mini example:

```r
de> dev.cur()
null device 
          1 
> par("bg")
[1] "transparent"
> dev.cur()
quartz 
     2 
> dev.off()
null device 
          1 
> dev.cur()
null device 
          1 
> dev.set(1)
null device 
          1 
> dev.cur()
quartz 
     2 
```

Looks as though I've come across this issue before in similar `with_device()`-like context:

https://github.com/rstudio/thematic/blob/c05ca534c7d1734fb9a782804980c998ff3457b3/R/hooks.R#L158-L169